lib/handlebars/compiler/javascript-compiler.js

```diff
@@ -426,7 +426,7 @@ JavaScriptCompiler.prototype = {
     if (this.isInline()) {
       this.replaceStack(current => [' != null ? ', current, ' : ""']);
 
-      this.pushSource(this.appendToBuffer(this.popStack()));
+      this.pushSource(this.appendToBuffer([this.popStack(), '+', "''"]));
     } else {
       let local = this.popStack();
       this.pushSource([
```

Comment:
What about non-inline statements (see below)? Do we also have to cast those to a string?

Reply:
Maybe it is better to add the empty string to "appendToBuffer", even though... I'll have another look. Something occurred to me

Reply:
I'll add discussion to the ticket. I think there might be a better solution, but I haven't grasped it completely yet

Reply:
I sided with your solution and pushed it.